controllers/set/status.go

```diff
@@ -8,34 +8,44 @@ import (
 	"github.com/metal-stack/metal-lib/pkg/pointer"
 )
 
+type FirewallConditionStatus struct {
+	IsReady, IsProgressing, IsUnhealthy bool
+}
+
+func evaluateFirewallConditions(fw *v2.Firewall, healthTimeout time.Duration) FirewallConditionStatus {
+	created := pointer.SafeDeref(fw.Status.Conditions.Get(v2.FirewallCreated)).Status == v2.ConditionTrue
+	ready := pointer.SafeDeref(fw.Status.Conditions.Get(v2.FirewallReady)).Status == v2.ConditionTrue
+	connected := pointer.SafeDeref(fw.Status.Conditions.Get(v2.FirewallControllerConnected)).Status == v2.ConditionTrue
+	seedConnected := pointer.SafeDeref(fw.Status.Conditions.Get(v2.FirewallControllerSeedConnected)).Status == v2.ConditionTrue
+	distance := pointer.SafeDeref(fw.Status.Conditions.Get(v2.FirewallDistanceConfigured)).Status == v2.ConditionTrue
+
+	allConditionsMet := created && ready && connected && seedConnected && distance
+	allocationTimeExceeded := created && time.Since(pointer.SafeDeref(fw.Status.MachineStatus).AllocationTimestamp.Time) < healthTimeout
+	unhealthyTimeExceeded := created && time.Since(pointer.SafeDeref(fw.Status.MachineStatus).AllocationTimestamp.Time) > healthTimeout
```

Comment:
I don't think you can use `AllocationTimestamp` because this will never change and at some point the health timeout will always be reached. Only the firewall-controller periodically renews an updated timestamp, so maybe you need to use this?

I would also be nice to disable the health timeout in case it's set to zero seconds.

Reply:
It's definitely an interesting idea. However, I think this would be an indirect implication of the firewall-controller not being able to reconcile anymore.

Reply:
If a firewall is set to healthy, it might happen that it goes unhealthy after a time, for example seed isn't connected anymore, right?

That means the first if case isn't a hit, but the second will always be a hit because after a certain time the timeout will be reached always like you said.
Even tho the firewall is unhealthy now, it is being set to progressing now always.

Isn't this a bug or what am I missing?

https://github.com/metal-stack/firewall-controller-manager/blob/91feb1f750de70a2008b81f3bf0dda41aa4be13f/controllers/set/status.go#L28-L39

Reply:
So using the firewall updated time instead of the machine allocation time?
`pointer.SafeDeref(fw.Status.ControllerStatus).Updated.Time`

Reply:
The problem is that pointer.SafeDeref(fw.Status.ControllerStatus).SeedUpdated.Time is no yet initalized when a deployment is created. 
Which leads to a deletion, before the deployment is initalized => SeedUpdated.Time defaults to  January 1, year 1, 00:00:00 UTC.